From bcc1c8b25c5e27b960d8ed78ee96342f8c436ea9 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Sun, 4 Mar 2018 04:07:27 -0600
Subject: [PATCH] "0001-Camera2-override-capture-support-level
 _______________________________

*Previous From 95c1b1fd6d4ca9486cc39fc3e30378f9f4711c02 Mon Sep 17 00:00:00 2001
*Previous From: Meticulus <theonejohnnyd@gmail.com>
*Previous Date: Mon, 5 Dec 2016 17:27:53 -0600
*Previous Subject: [PATCH] Camera2: override capture support level
*
*-The capture support level detected for our Camera is
*LIMITED_YUV which leads to incorrect picture sizes. Well,
*they might be for YUV but our Cameras are capable of
*LIMITED_JPEG which corrects the camera issues. GservicesHelper
*seems to be a place where this detection can be overriden using
*a ContentResolver. I thought it would be simpler to just set
*a property.
*
*-CaptureSupportLevel is an enum with these values:
*ZSL = 1
*LEGACY_JPEG = 2
*LIMITED_JPEG = 3
*LIMITED_YUV = 4
*and you can use -1 for unused. The property should correspond
*to one of these values.
*
*Change-Id: I72ad06572ecc6fbc7b016f53d2e8a0f50cbe3b6a"

Change-Id: I025a5f81a06fd5190e478512665dc87392655751
---
 src_pd/com/android/camera/util/GservicesHelper.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src_pd/com/android/camera/util/GservicesHelper.java b/src_pd/com/android/camera/util/GservicesHelper.java
index 12ae56a..2ff9828 100644
--- a/src_pd/com/android/camera/util/GservicesHelper.java
+++ b/src_pd/com/android/camera/util/GservicesHelper.java
@@ -38,11 +38,11 @@ public class GservicesHelper {
     }
 
     public static int getCaptureSupportLevelOverrideBack(ContentResolver contentResolver) {
-        return -1;
+        return 3;
     }
 
     public static int getCaptureSupportLevelOverrideFront(ContentResolver contentResolver) {
-        return -1;
+        return 3;
     }
 
     public static int getMaxAllowedNativeMemoryMb(ContentResolver contentResolver) {
-- 
2.7.4

